api/src/main/java/org/openmrs/BaseCustomizableData.java

```diff
@@ -9,124 +9,46 @@
  */
 package org.openmrs;
 
-import java.util.ArrayList;
 import java.util.Collection;
 import java.util.LinkedHashSet;
-import java.util.List;
-import java.util.Set;
 
 import org.openmrs.attribute.Attribute;
-import org.openmrs.customdatatype.CustomValueDescriptor;
 import org.openmrs.customdatatype.Customizable;
 
 /**
- * Extension of {@link BaseOpenmrsData} for classes that support customization via user-defined attributes.
+ * Extension of {@link BaseOpenmrsData} for classes that support customization via user-defined
+ * attributes.
+ * 
  * @param <A> the type of attribute held
  * @since 1.9
  */
 public abstract class BaseCustomizableData<A extends Attribute> extends BaseOpenmrsData implements Customizable<A> {
 	
-	private Set<A> attributes = new LinkedHashSet<A>();
+	private Collection<A> attributes = new LinkedHashSet<A>();
 	
 	/**
 	 * @see org.openmrs.customdatatype.Customizable#getAttributes()
 	 */
 	@Override
-	public Set<A> getAttributes() {
+	public Collection<A> getAttributes() {
```

Comment:
Well, since I'm new to the project you have to decide ;) 

The thing is, we can't have two methods with the same name but different return-type in Java. So deprecation would be difficult. Since Collection is the super-type of Set, I think its unlikely to break existing code. But I'm not really sure, since I don't know the project well.